packages/lesswrong/server/resolvers/dialogueChecksResolvers.ts

```diff
@@ -4,34 +4,142 @@ import { augmentFieldsDict } from "../../lib/utils/schemaUtils";
-  const match = await getMatchingDialogueCheck(dialogueCheck);
-  if (match) {
-    await createNotifications({
-      userIds: [dialogueCheck.userId],
-      notificationType: "newDialogueMatch",
-      documentType: "dialogueCheck",
-      documentId: dialogueCheck._id,
-    });
-    await createNotifications({
-      userIds: [match.userId],
-      notificationType: "newDialogueMatch",
-      documentType: "dialogueCheck",
-      documentId: match._id,
-    });
+import { getUser } from '../../lib/vulcan-users/helpers';
+import { getAdminTeamAccount } from '../../server/callbacks/commentCallbacks.ts'
+import {SyncPreference} from "../../lib/collections/dialogueMatchPreferences/schema.ts";
+
+async function notifyUsersMatchingDialogueChecks (dialogueCheck: DbDialogueCheck, match: DbDialogueCheck, associatedMessage?: DbMessage) {
+  await createNotifications({
+    userIds: [dialogueCheck.userId],
+    notificationType: "newDialogueMatch",
+    documentType: "dialogueCheck",
+    documentId: dialogueCheck._id,
+    extraData: {associatedMessage}
+  });
+  await createNotifications({
+    userIds: [match.userId],
+    notificationType: "newDialogueMatch",
+    documentType: "dialogueCheck",
+    documentId: match._id,
+    extraData: {associatedMessage}
+  });
+}
+
+async function messageUsersMatchingDialogueChecks (
+  userId: string,
+  targetUserId: string,
+  topicNotes: string,
+  formatSync: boolean,
+  formatAsync: boolean,
+  formatNotes: string
+) {
+  const lwAccount = await getAdminTeamAccount();
+  const currentUser = await getUser(userId);
+  const targetUser = await getUser(targetUserId);
+
+  // Create a new conversation with both users
+  const conversationData = {
+    participantIds: [userId, targetUserId, lwAccount._id],
+    title: `Dialogue Match between ${currentUser?.displayName} and ${targetUser?.displayName}!`
+  }
+
+  const conversation = await createMutator({
+    collection: Conversations,
+    document: conversationData,
+    currentUser: lwAccount,
+    validate: false,
+  });
+
+  let messageContents =
+    `<div>
+      <p>You two have matched via Dialogue Matching! You can now message each other to brainstorm potential dialogue topics or set up a time to talk.</p>
+      <p>For some ideas of conversation topics, feel free to look at <a href="https://www.lesswrong.com/posts/hc9nMipTXy2sm3tJb/vote-on-interesting-disagreements">this list of interesting disagreements</a>.</p>
+
+      <p>Notes from ${currentUser?.displayName}:</p>
+      ${topicNotes ? `<p>On topics: "${topicNotes}"</p>` : ''}
+      ${formatSync ? `<p>• Up for sync</p>` : ''}
+      ${formatAsync ? `<p>• Up for async</p>` : ''}
+      ${formatNotes ? `<p>Format notes: "${formatNotes}"</p>` : ''}
+    </div>`
+
+  // Add a message to the conversation
+  const messageData = {
+    userId: lwAccount._id,
+    contents: {
+      originalContents: {
+        type: "html",
+        data: messageContents
```

Comment:
This message content html is unsafe as it is constructed directly from user inputs without sanitation. This is how the AI kills your website.